vim/core/oni-core-interop/plugin/init.vim

```diff
@@ -19,27 +19,34 @@ function OniNotifyBufferUpdate()
 
     if b:changedtick > b:last_change_tick
         let b:last_change_tick = b:changedtick
-        let buffer_lines = getline(1, '$')
-        let context = OniGetContext()
-        call OniNotify(["buffer_update", context, buffer_lines])
+        if mode() == 'i'
```

Comment:
Yeah, the last_cursor_line business was an attempt to handle more complex things like multiline deletes and such, but It's a bit hard to make all the scripts